app/contracts/placeholder_users/base_contract.rb

```diff
@@ -0,0 +1,57 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+require 'model_contract'
+
+module PlaceholderUsers
+  class BaseContract < ::ModelContract
+    attribute :type
+    attribute :lastname,
+              writeable: ->(*) { user.admin? }
+    attribute :name,
+              writeable: ->(*) { user.admin? }
```

Comment:
I agree.

Reply:
Hmm, removing `:lastname` here makes other specs fail. In the `PlaceholderUser` model we define `alias_attribute(:name, :lastname)`. Apparently the SetAttributesService detects that also `lastname` gets updated if `name` is changed and reject the change unless we also allow `lastname` to be writable. I will leave it in unless you know of a better solution.

Reply:
You could try using 

`attribute_alias :lastname, :name`

It would still have the DB name in the contract but at least the connection between the two is made visible.

See: https://github.com/opf/openproject/blob/dev/app/contracts/base_contract.rb#L66-L70
And https://github.com/opf/openproject/blob/dev/modules/grids/app/contracts/grids/base_contract.rb#L45

Reply:
When I add `attribute_alias :lastname, :name` I still need to declare the `:lastname` attribute as writable for admins. Or least I don't get it to work otherwise.